### PR TITLE
Integrate k8s

### PR DESCRIPTION
**Pull request by nemo83**
Integrated delta to k8s api to fetch applications deployment states

* added `aws-iam-authenticator` to dockerfile to allow Kube Client authentication
* The `kube-config.yaml` is used by the K8s Java client to initialize itself

### CODE REVIEW

File: api/app/actors/K8sBuildActor.scala [hunk 1]
```diff
@@ -66,17 +68,13 @@ class K8sBuildActor @javax.inject.Inject() (
...
-    }
-  */
+    val versions = kubernetesService.getDeployedVersions(build.name)
+    buildLastStatesDao.upsert(
+      usersDao.systemUser,
+      build,
+      StateForm(versions = versions)
+    )
+    StateFormatter.label(versions)
     println(s"TODO: captureLastState for build '${build.id}''")
```
Comment (mbryzek) [inline api/app/actors/K8sBuildActor.scala]: let's remove the println which was here only to touch the build variable

File: api/conf/base.conf [hunk 2]
```diff
@@ -158,4 +159,6 @@ rollbar-actor-context {
   }
 }
 
+kube.config.path = "~/.kube/config"
```
Comment (mbryzek) [inline api/conf/base.conf]: worried this won't resolve in production

Comment (nemo83) [reply api/conf/base.conf]: Fixed already. Tests weren't passing, In dev this should be ok.

File: api/app/lib/Bindings.scala [hunk 3]
```diff
@@ -19,10 +19,10 @@ class KubernetesModule extends Module {
   def bindings(env: Environment, conf: Configuration) = {
     env.mode match {
       case Mode.Prod | Mode.Dev => Seq(
-        bind[DefaultKubernetesService].to[KubernetesService]
+        bind[KubernetesService].to[DefaultKubernetesService]
       )
       case Mode.Test => Seq(
-        bind[DummyK8sService].to[KubernetesService]
+        bind[KubernetesService].to[DummyK8sService]
```
Comment (mbryzek) [inline api/app/lib/Bindings.scala]: Can we rename dummy to mock?

**Pull request merged**